src/HaskellCI/OptparseGrammar.hs

```diff
@@ -25,14 +25,14 @@ import qualified Options.Applicative         as O
 import HaskellCI.OptionsGrammar
 
 data SomeParser s where
-    SP :: (Maybe String -> Maybe String -> O.Parser (s -> s)) -> SomeParser s
+    SP :: (Maybe MetaVar -> Maybe BashCompletionAction -> Maybe Help -> O.Parser (s -> s)) -> SomeParser s
```

Comment:
Use `Maybe Completer` and `bashCompleter` to defined the `metaActionHelp`.

If someone feels like adding completers for other arguments, the support would be there. Also less `String`s.

Reply:
Relatedly, does `bashCompleter` work with `zsh`? The `optparse-applicative` docs are sparse on the topic.

Reply:
Good, switched to `Completer`.

I haven't used `zsh` before but dived into it and installed it.  Got completion with zsh to work, it is analogous to bash completion.  The completion script is generated by
```
$ haskell-ci --zsh-completion-script `which haskell-ci`
```
and (by convention) to be placed in a file `_haskell-ci` somewhere on zsh's `FPATH`.  That did it for me.